r/R/dplyr-functions.R

```diff
@@ -330,6 +330,39 @@ arrow_string_join_function <- function(null_handling, null_replacement = NULL) {
   }
 }
 
+# Currently, Arrow does not supports a locale option for string case conversion
+# functions, contrast to stringr's API, so the 'locale' argument is only valid
+# for stringr's default value ("en"). The following are string functions that
+# take a 'locale' option as its second argument:
+#   str_to_lower
+#   str_to_upper
+#   str_to_title
+#
+# Arrow locale will be supported with ARROW-14126
+nse_funcs$str_to_lower <- function(string, locale = "en") {
+  if (!identical(locale, "en")) {
+    stop("Providing 'locale' to 'str_to_lower' is not supported in Arrow; ",
+    "to change locale use 'Sys.setlocale()'", call. = FALSE)
```

Comment:
Small phrasing suggestion (would need copying to the others too).

```suggestion
    stop("Providing a value for 'locale' other than the default ('en') is not supported by Arrow. ",
    "To change locale, use 'Sys.setlocale()'", call. = FALSE)
```

Reply:
I like Nic's phrasing here, seeing it repeated three times: should we write/store the message once and use it in each `stop` so we know the same is being used. If we do want to keep the function name in it (I don't think we need to, but if we do) we could make a small helper that plots that in.